animation_nodes/data_structures/midi/midi_note.py

```diff
@@ -8,18 +8,34 @@ class MIDINote:
     timeOff: float = 0
     velocity: float = 0
 
-    def evaluate(self, time, attackTime, attackInterpolation, releaseTime, releaseInterpolation):
-        peakTime = self.timeOn + attackTime
-        endTime = self.timeOff + releaseTime
-        if self.timeOff >= time >= peakTime:
-            return 1
-        elif peakTime > time >= self.timeOn:
-            if attackTime == 0: return 1
-            return attackInterpolation((time - self.timeOn) / attackTime)
-        elif endTime >= time > self.timeOff:
-            if releaseTime == 0: return 1
-            return releaseInterpolation(1 - ((time - self.timeOff) / releaseTime))
-        return 0.0
+    def evaluate(self, time, attackTime, attackInterpolation, decayTime, decayInterpolation, sustainLevel, 
+        releaseTime, releaseInterpolation, velocitySensitivity):
+
+        def valueInEnvelope():
```

Comment:
Prefer not to have local functions. Just pull this function outside into a separate method and call it `evaluateEnvelop`.

Reply:
done